hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -168,7 +168,8 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       if (tbl != null) {
         // If we try to create the table but the metadata location is already set, then we had a concurrent commit
         if (base == null && tbl.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP) != null) {
-          throw new AlreadyExistsException("Table already exists: %s.%s", database, tableName);
```

Comment:
Well, replacing `AlreadyExistsException` with `CommitFailedException` may lead to unexpected consequences.

Reply:
Agreed, I think `AlreadyExistsExpection` is the correct exception for the problem and we should raise as much info as possible accordingly.